foxglove_bridge_base/include/foxglove_bridge/websocket_server.hpp

```diff
@@ -381,24 +386,30 @@ inline void Server<ServerConfiguration>::handleConnectionOpened(ConnHandle hdl)
                  })
               .dump());
 
-  std::vector<Channel> channels;
-  for (const auto& [id, channel] : _channels) {
-    (void)id;
-    channels.push_back(channel);
+  {
+    std::shared_lock<std::shared_mutex> lock(_channelsMutex);
+    std::vector<Channel> channels;
+    for (const auto& [id, channel] : _channels) {
+      (void)id;
+      channels.push_back(channel);
+    }
+    sendJson(hdl, {
+                    {"op", "advertise"},
+                    {"channels", std::move(channels)},
+                  });
```

Comment:
Can this be modified to move the `sendJson` call outside of the lock? Declare `std::vector<Channel> channels;` above and only populate it inside the lock.

Reply:
yap, that makes sense. Will do